BUILD.bazel

```diff
@@ -0,0 +1,57 @@
+java_plugin(
+    name = "auto_value_plugin",
+    processor_class = "com.google.auto.value.processor.AutoValueProcessor",
+    deps = ["@com_google_auto_value_auto_value//jar"],
+)
+
+java_library(
+    name = "auto_value",
+    exported_plugins = [":auto_value_plugin"],
+    exports = ["@com_google_auto_value_auto_value//jar"],
```

Comment:
Does it make sense to move this `exports` to the `"auto_value_plugin"` target?

(Just learning about `exports` and `java_plugin`--I've not run into them before)

Reply:
This was the recommended way to do it... I'm not sure the reason for your suggestion, do you see a problem with it how it is?

Reply:
I'm going to merge because it works as-is; we can discuss offline if you think something should change later.